app/src/main/java/com/example/android/sunshine/app/LocationEditTextPreference.java

```diff
@@ -52,6 +62,34 @@ private void init(Context context, AttributeSet attrs) {
                 a.recycle();
             }
         }
+
+        GoogleApiAvailability apiAvailability = GoogleApiAvailability.getInstance();
+        int resultCode = apiAvailability.isGooglePlayServicesAvailable(context);
+        if (resultCode == ConnectionResult.SUCCESS) {
```

Comment:
perhaps this code could be wrapped in a collaborator - that way you could potentially swap google places with yahoo places or whatever you want 😉